modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -979,9 +1064,17 @@ private CompletableFuture<Object> processSingleEntryAction(ReadWriteSingleRowRep
                 return lockFut.thenCompose(lockedRowId -> {
                     boolean inserted = lockedRowId == null;
 
-                    CompletableFuture raftFut =
+                    CompletableFuture<Object> raftFut =
                             lockedRowId == null ? applyCmdWithExceptionHandling(
-                                    new UpdateCommand(commitPartitionId, new RowId(partId), searchRow, txId)) :
+                                    msgFactory.updateCommand()
```

Comment:
Talking about methods extractions, should we create private methods for all these builds? Could it simplify the code?

Reply:
Done, saved a lot of lines of code.